### PR TITLE
[SPARK-35822][UI] Spark UI-Executor tab is empty in IE11

### PR DESCRIPTION
**Pull request by dgd-contributor**
### What changes were proposed in this pull request?
Refactor some functions in utils.js to fix the empty UI-Executor tab in yarn mode in IE11.

### Why are the changes needed?
Spark UI-Executor tab is empty in IE11: So this PR to fix this.
Executortab_IE

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existed UT Testcase

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @sarutak @gengliangwang FYI

File: core/src/main/resources/org/apache/spark/ui/static/utils.js [hunk 1]
```diff
@@ -227,4 +227,12 @@ function createRESTEndPointForMiscellaneousProcess(appId) {
   }
   return uiRoot + "/api/v1/applications/" + appId + "/allmiscellaneousprocess";
 }
+
+function getBaseURI(){
+  let baseURI = document.baseURI;
+  if(!baseURI){
```
Comment (sarutak) [inline core/src/main/resources/org/apache/spark/ui/static/utils.js:233]:
```suggestion
  if (!baseURI){
```

Or, simply can we simply do `document.baseURI || document.URL`?

File: core/src/main/resources/org/apache/spark/ui/static/utils.js [hunk 2]
```diff
@@ -227,4 +227,12 @@ function createRESTEndPointForMiscellaneousProcess(appId) {
   }
   return uiRoot + "/api/v1/applications/" + appId + "/allmiscellaneousprocess";
 }
+
+function getBaseURI(){
+  let baseURI = document.baseURI;
```
Comment (sarutak) [inline core/src/main/resources/org/apache/spark/ui/static/utils.js:232]: This change seems to aim IE11. Is it support EC6 features? (`let` here).

**Comment by sarutak** [comment]
@dgd-contributor Thank you for fixing this issue. Could you paste screenshots on the description?

File: core/src/main/resources/org/apache/spark/ui/static/utils.js [hunk 3]
```diff
@@ -227,4 +227,12 @@ function createRESTEndPointForMiscellaneousProcess(appId) {
   }
   return uiRoot + "/api/v1/applications/" + appId + "/allmiscellaneousprocess";
 }
+
+function getBaseURI(){
```
Comment (sarutak) [inline core/src/main/resources/org/apache/spark/ui/static/utils.js:231]:
```suggestion
function getBaseURI() {
```

**Comment by gengliangwang** [comment]
I run `git grep "document.baseURI"` and get the following:
```
core/src/main/resources/org/apache/spark/ui/static/log-view.js:  var words = document.baseURI.split('/');
core/src/main/resources/org/apache/spark/ui/static/stagepage.js:  var queryString = document.baseURI.split('?');
core/src/main/resources/org/apache/spark/ui/static/stagepage.js:  var words = document.baseURI.split('/');
core/src/main/resources/org/apache/spark/ui/static/stagepage.js:  var words = document.baseURI.split('?');
core/src/main/resources/org/apache/spark/ui/static/utils.js:  var words = document.baseURI.split('/');
core/src/main/resources/org/apache/spark/ui/static/utils.js:  var words = document.baseURI.split('/');
core/src/main/resources/org/apache/spark/ui/static/utils.js:  var words = document.baseURI.split('/');
core/src/main/resources/org/apache/spark/ui/static/utils.js:  var words = document.baseURI.split('/');
```

Shall we fix all of them?

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you, @dgd-contributor . If the patch is tested in Chrome/Safari too, the approach looks reasonable.

BTW, to the committers, if you want to have this on `branch-3.2` (or olders), please change the JIRA issue types first. Currently,
- SPARK-35821 is filed as `New Feature` Umbrella issue.
- SPARK-35822 is a subtask of it.

So, according to the AS-IS status, this PR should not land to the release branch. This is for Apache Spark 3.3 (master) branch.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
The type of SPARK-35821 is changed to Umbrella, we can merge this one to 3.2

**Review by sarutak — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you all!
Merged to master/3.2.

**Comment by dongjoon-hyun** [comment]
I landed to master/3.2 only. For the other release branches, we can revisit after Apache Spark 3.2.0 release.

**Pull request closed without merging**